src/casp/services/api/data_manager/cellarium_general.py

```diff
@@ -143,16 +143,25 @@ def log_user_activity(
 
     def get_remaining_quota_for_user(self, user: models.User) -> int:
         """
-        Get remaining cell processing quota for a specific user (i.e. their quota minus cells processed this week)
+        Get remaining cell processing quota for a specific user (i.e. their quota minus cells processed this week
+        or their remaining lifetime quota if they have one)
 
         :param user: User object to check quota for
 
         :return: Remaining cell processing quota
         """
-        # Return the remaining quota, or 0 if it's negative (which might happen if the user exceeds
-        # their quota, either because we allow them to, or because of any weirdness in the
-        # simultaneous processing of cells)
-        return max(user.cell_quota - self.get_cells_processed_this_week_for_user(user=user), 0)
+        # Return the smaller of the remaining weekly and lifetime quotas, or 0 if it's negative
+        # (which might happen if the user exceeds their quota, either because we allow them to, or
+        # because of any weirdness in the simultaneous processing of cells)
+        remaining_quota = min(
+            user.cell_quota - self.get_cells_processed_this_week_for_user(user),
+            (
+                user.lifetime_cell_quota - user.total_cells_processed
+                if user.lifetime_cell_quota is not None
+                else float("inf")
+            ),
+        )
+        return max(remaining_quota, 0)
```

Comment:
nit, but I think it could be more readable if you break it into steps like this:
```python3
weekly_remaining_quota = user.cell_quota - self.get_cells_processed_this_week_for_user(user=user)

lifetime_remaining_quota = (
    user.lifetime_cell_quota - user.total_cells_processed
    if user.lifetime_cell_quota is not None
    else float("inf")
)

remaining_quota = min(weekly_remaining_quota, lifetime_remaining_quota)

# Ensure the remaining quota is non-negative
return max(remaining_quota, 0)
```

Reply:
Just a thought: it might make more sense to move the quota calculation into a service method rather than keeping it in the data manager. This could better align with the separation of concerns principle.

Reply:
I think I had originally started doing it this way when I was writing the original quota code, but I realized both the cell operations service and the cellarium general service need to use this method, so I wasn't sure if it made sense to move it into one or the other.  I guess maybe we could make a new file to just put the quota stuff in?  What do you think?

Reply:
I like this idea. In the future, it will only help us better organize and navigate the code!
I think it might make sense to unify this into another "service" class, like a quota service or something similar.